src/github/index.ts

```diff
@@ -0,0 +1,67 @@
+import * as Octokit from '@octokit/rest'
+import { addLogging, Logger  } from './logging'
+import { addPagination } from './pagination'
+import { addRateLimiting } from './rate-limiting'
+
+import { addGraphQL } from './graphql'
+
+/**
+ * the [@octokit/rest Node.js module](https://github.com/octokit/rest.js),
+ * which wraps the [GitHub API](https://developer.github.com/v3/) and allows
+ * you to do almost anything programmatically that you can do through a web
+ * browser.
+ * @typedef github
+ * @see {@link https://github.com/octokit/rest.js}
+ */
+export const EnhancedGitHubClient = function (options) {
+  const octokit = <OctokitWithPagination> new Octokit(options)
+
+  addRateLimiting(octokit, options.limiter)
+  addLogging(octokit, options.logger)
+  addPagination(octokit)
+  addGraphQL(octokit)
+
+  return octokit
+}
+
+export interface Options extends Octokit.Options {
+  debug: boolean
+  logger: Logger
+  limiter?: any
+}
+
+export interface OctokitRequestOptions {
+  method: string
+  url: string
+  headers: any
+  query?: string
+  variables?: Variables
+}
+
+export interface OctokitResult {
+  meta: {
+    status: string
+  }
+}
+
+export interface OctokitError {
+  code: number
+  status: string
+}
+
+export interface OctokitWithPagination extends Octokit {
+  paginate: (res: Promise<Octokit.AnyResponse>, callback: (results: Octokit.AnyResponse) => void) => void
```

Comment:
returns `Promise<any[]>`, not `void`

Reply:
in fact it should probably be a generic type to avoid `any`:
```ts
export interface OctokitWithPagination<T> extends Octokit {
  paginate: (res: Promise<Octokit.AnyResponse>, callback: (results: Octokit.AnyResponse) => void) => Promise<T[]>
```